src/Chapter3.hs

```diff
@@ -1006,6 +1142,189 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
+-- Couldn't make it simpler :(
+data KnightAction
+  = KnightAttack
+  | Drink
+  | Cast
+  deriving (Show)
+
+data MonsterAction
+  = MonsterAttack
+  | Run
+  deriving (Show)
+
+data Knight = Knight
+  { knightHealth :: Int
+  , knightAttack :: Int
+  , knightDefense :: Int
+  , knightActions :: [KnightAction]
+  , baseKnightHealth :: Int
+  , baseKnightHealthIncrease :: Int
+  , baseKnightDefense :: Int
+  , baseKnightDefenseIncrease :: Int
```

Comment:
What `base` meant here?

Reply:
It's for making heal and buff diminish with each application. 'Health' is current health, and 'base' is the number, which affects the calculations: health cannot go over the base value with healing, and the further current defense is from base, the less it's increased.

Reply:
Thanks for clarifications! I see now 👍🏼 

One way to document it in Haskell is to add the comment to each field, e.g.:
```haskell
  , baseKnightHealth :: Int  -- ^ The maximum amount of Health for the 'Knight"
```